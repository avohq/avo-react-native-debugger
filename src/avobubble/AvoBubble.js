import React from 'react';
import { View, Image, Text, Platform, ImageBackground } from 'react-native';
import { images } from '../images';
import { colors } from '../colors';
import PropTypes from 'prop-types';
import { styles } from './AvoBubbleStyles';

export default function AvoBubble(props) {

    let mainImage = images.avo_buuble;
    if (props.hasErrors) {
        mainImage = images.avo_bubble_error;
    }

    let badgeBackground = images.badge_green;
    if (props.hasErrors) {
        badgeBackground = images.badge_white;
    }

    let textColor = colors.background;
    if (props.hasErrors) {
        textColor = colors.error;
    }

    if (props.newItems > 0) {
        let itemsCounter = props.newItems;
        if (Platform.OS === 'android' && itemsCounter > 999) {
            itemsCounter = "...";
        }
        return (
            <View style={styles.mainImage}>
                <Image source={mainImage} style={styles.mainImage} />
                <ImageBackground style={styles.badge} source={badgeBackground}>
                    <Text adjustsFontSizeToFit={Platform.OS === 'ios'}
                        ellipsizeMode={'tail'} numberOfLines={1}
                        style={[styles.messageCounter, { color: textColor }]}>{itemsCounter}</Text>
                </ImageBackground>
            </View>
        );
    } else {
        return (
            <View>
                <Image source={mainImage} style={styles.mainImage} />
            </View>
        );
    }
}

AvoBubble.propTypes = {
    newItems: PropTypes.number,
    hasErrors: PropTypes.bool
}