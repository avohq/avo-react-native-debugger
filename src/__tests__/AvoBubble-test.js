import React from "react";
import renderer from "react-test-renderer";

import AvoBubble from "../avobubble/AvoBubble";

describe("AvoBubble", () => {
  test("Renders correctly", () => {
    const tree = renderer
      .create(<AvoBubble newItems={4} hasErrors={false} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("Renders correctly with error", () => {
    const tree = renderer
      .create(<AvoBubble newItems={4} hasErrors={true} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("Renders correctly without new items", () => {
    const tree = renderer.create(<AvoBubble newItems={0} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("Renders correctly without props set", () => {
    const tree = renderer.create(<AvoBubble />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
