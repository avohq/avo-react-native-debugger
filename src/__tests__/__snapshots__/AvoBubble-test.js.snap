// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvoBubble Renders correctly 1`] = `
<View
  style={
    Object {
      "height": 50,
      "width": 50,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/img/avo_bubble.png",
      }
    }
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "height": 24,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "width": 24,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/badge_green.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 24,
            "width": 24,
          },
          undefined,
        ]
      }
    />
    <Text
      adjustsFontSizeToFit={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 9,
            "paddingBottom": 4,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      4
    </Text>
  </View>
</View>
`;

exports[`AvoBubble Renders correctly with error 1`] = `
<View
  style={
    Object {
      "height": 50,
      "width": 50,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/img/avo_bubble_error.png",
      }
    }
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "height": 24,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "width": 24,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/badge_white.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 24,
            "width": 24,
          },
          undefined,
        ]
      }
    />
    <Text
      adjustsFontSizeToFit={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontSize": 9,
            "paddingBottom": 4,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          Object {
            "color": "#D94553",
          },
        ]
      }
    >
      4
    </Text>
  </View>
</View>
`;

exports[`AvoBubble Renders correctly without new items 1`] = `
<View>
  <Image
    source={
      Object {
        "testUri": "../../../src/img/avo_bubble.png",
      }
    }
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
</View>
`;

exports[`AvoBubble Renders correctly without props set 1`] = `
<View>
  <Image
    source={
      Object {
        "testUri": "../../../src/img/avo_bubble.png",
      }
    }
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
</View>
`;
